pyvista/plotting/actor_properties.py

```diff
@@ -0,0 +1,149 @@
+"""Module containing pyvista implementation of vtkProperty."""
+from enum import Enum
+
+import pyvista as pv
+
+
+class ShaderModel(Enum):
+    """Types of shader methods available."""
+
+    FLAT = 0
+    GOURAUD = 1
+    PHONG = 2
+    PBR = 3
+
+
+class ReprModel(Enum):
+    """Types of representations the models can have."""
+
+    POINTS = 0
+    WIREFRAME = 1
+    SURFACE = 2
+
+
+class ActorProperties:
+    """Properties wrapper for ``vtkProperty``. Contains the surface properties of the object.
+
+    Parameters
+    ----------
+    properties : pv._vtk.vtkProperty
+        VTK properties of the current object.
+
+    Examples
+    --------
+    Access the properties of the z axis shaft.
+
+    >>> import pyvista as pv
+    >>> axes = pv.Axes()
+    >>> z_axes_prop = axes.axes_actor.z_axis_shaft_properties
+    >>> z_axes_prop.color = (1, 1, 0)
+    >>> z_axes_prop.opacity = 0.5
+    >>> axes.axes_actor.SetShaftTypeToCylinder()
+    >>> pl = pv.Plotter()
+    >>> pl.add_actor(axes.axes_actor)   # doctest:+SKIP
+    >>> pl.add_mesh(pv.Sphere())   # doctest:+SKIP
+    >>> pl.show()   # doctest:+SKIP
+    """
+
+    def __init__(self, properties: pv._vtk.vtkProperty) -> None:
+        super().__init__()
+        self.properties = properties()
```

Comment:
This is really odd to me. Why is a bound method often passed to this constructor and not the object itself? The type annotation implies that a `vtkProperty` instance should be passed but this is used where a bound method that produces that instance is passed.

Is this because of being unable to wrap/shallow copy `vtkProperty`?

Reply:
I was having issues with maintaining the information I set in the properties while doing this. Passing the object works fine, it is solved now.